msbuild/Xamarin.Mac.Tasks.Core/Tasks/DetectSdkLocations.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.MacDev.Tasks;
+using Xamarin.MacDev;
+
+namespace Xamarin.Mac.Tasks
+{
+	public class DetectSdkLocationsTaskBase : Task
+	{
+		#region Inputs
+
+		public string SessionId { get; set; }
+
+		// This is also an input
+		[Output]
+		public string SdkVersion {
+			get; set;
```

Comment:
Agreed with all the formatting comments, but this is a copy paste from the original and I didn't want to change it. See here: https://github.com/xamarin/xamarin-macios/pull/77/files#diff-2e650c9beeab2092efcab1aea606b5e6L19